### ISSUE TITLE
Open Computer recipes

### ISSUE THREAD

**Issue by basdxz** [GTNewHorizons/NewHorizons#727]
Some of the Gregifed in interesting ways. Most of the expansion cards, ram and some other thing should IMO be crafted in an assembler w/ soldering alloy/lead/tin ect. Also why does it use it's own proprietary circuit boards and not Gregtech ones? Heck the wires don't need a laser engraver.
Oh and the Open Computer Assembler should use an assembler and forming press. Along with the HDD should need some form of shape to make the disk and not just 4 aluminum plates in a crafting table. Also why does the CPU use screws and not thermal paste?

**Comment by Dream-Master** [comment]
Open Computer has his own recipes by default. I need to go over it.

**Comment by bigbass1997** [comment]
The recipes @basdxz is talking about have already been "gregified", or in other words, they are not the original OpenComputers default recipes. Though not all of the OpenComputers' recipes have been changed. However the current gregified recipes don't make a lot of sense. Such as requiring screws to make a CPU.

Note that the Memory aka RAM items inherently require the Assembler. The Tier 1 Memory needs the assembler, and then all higher tier memory needs Tier 1 Memory.

Various parts, such as the Microchips and Transistors require the assembler. But Care Bases and most Cards (e.g. Graphics Cards) do not use the assembler except for the parts that makeup the cards.

Bottom line: Recipes for OpenComputers have been changed, but only to an extent. Some recipes have just added some GT tools, while others have been completely changed. And then some recipes that have been changed, such as the CPU or Disk Platters, are not entirely realistic.

**Comment by Dream-Master** [comment]
I change only crafting recipe for Computer Case Tier 2,  Computer Case Tier 3 and the Server Rack. The Printed Circuit Board needs a Assembler. Yes the whole stuff needs to be redone. My problem is the assembler con onl combine 2 items and a fluid. Some times you need a lot more of components. The only way i can add it is with the Assembly Machine but this is at high tier only possible.

**Comment by Dream-Master** [comment]
New version is up will check the recipe when i nerf this mod.

**Comment by ianespana** [comment]
I think one other recipe that should be looked at is the T1 CPU. For some reason it requires HV, while the rest of the stuff to make the basic computer needs MV. That's a little bit confusing, and IMO the T1 CPU should be accessible in MV (MV bein the entry point for OC)

**Comment by repo-alt** [comment]
When/if you will look at OC, please add the oredcited recipes for T1 and T2 microchips, because now they require ic2 circuts. While this is ok for manual crafting (GT assembler treats all circuits of the same  tier as equal), it prohibits crafting e.g. T1 microchips using microprocessors via AE autocrafting, because AE interface <> assembler interaction does not perform necessary conversion magic. (T3 microchip recipe uses several variants of the circuits, which is correct)

**Comment by Dream-Master** [comment]
@Henny22 any news here

**Comment by Henny022** [comment]
some robot upgrades are missing, but otherwise all recipes are done.
then i only need to adjust the voltages of the recipes and redo the integrated circuits to solve some conflicts.
regarding the oredicted circuits @repo-alt mentioned, is there a way to archive that, without iterating over all the possible circuits?

**Comment by Dream-Master** [comment]
@Henny022 let me know when you done.

**Comment by repo-alt** [comment]
As far as I understand, to make the recipe oredicted you don't need to enumerate anything. Look at the T2 microchip recipe in config/opencomputers/gregtech.recipes and compare it with the T3 recipe:
chip2 {
  type: gt_assembler
  input: ["ic2.itemPartCircuitAdv", "oc:materialTransistor"]
  count: [1, 8]
  eu: 25
  time: 640
  output: 4
}
if here "ic2.itemPartCircuitAdv" would be replaced by 'circuitAdvanced' (w/o quotes), it will use ore dictionary.

**Comment by Henny022** [comment]
i make the recipes in java with the gt api, because there was a problem when using the opencomputers config, i dont remember which one anymore (aside from one mistyping breaking the entire file making debuging SUPER hard)

**Comment by Henny022** [comment]
you can look at my fork of the gtnh core mod, to see my new recipes

**Comment by repo-alt** [comment]
I can see your fork but I don't see any of your commits in it, can you post a link?

**Comment by Henny022** [comment]
https://github.com/Henny022/NewHorizonsCoreMod/blob/OC-script/src/main/java/com/dreammaster/gthandler/GT_MachineRecipeLoader.java#L1228
and
https://github.com/Henny022/NewHorizonsCoreMod/blob/OC-script/src/main/java/com/dreammaster/gthandler/GT_MachineRecipeLoader.java#L1406
are the code blocks that handle OpenComputers

**Comment by repo-alt** [comment]
Oh, somehow I managed to miss the 'OC-Script' branch. Thanks.
Looks like the recipes there are using ore dictionary (GT_OreDictUnificator.get(OrePrefixes.circuit, Materials.Advanced, 1)), so there should be no issues with autocrafting.

**Comment by Dream-Master** [comment]
@Henny022 so i need an update here how long did you need to do recipes here ready?

**Comment by Henny022** [comment]
@repo-alt the GT_OreDictUnificator only returns the first item out of the Oredict
@Dream-Master i started the branch 8th of august but had to do different stuff in between

**Comment by repo-alt** [comment]
@Henny022 Indeed, I read java as if it was a minetweaker script.
I looked at OC recipe registration and they actually enumerate all the oredicted entries in recipes registration. Look e.g at https://github.com/MightyPirates/OpenComputers/blob/OC1.5-MC1.7.10/src/main/scala/li/cil/oc/integration/gregtech/RecipeHandler.scala, parseRecipe and parseIngredientList

**Comment by Dream-Master** [comment]
@Henny022 i guess i do recipe by my own to get it faster :P

**Comment by Dream-Master** [comment]
Recipes added only Computronics are missing
